how-gazelle-works.md

```diff
@@ -85,3 +85,182 @@ gazelle_binary(
     ],
 )
 ```
+
+## Manipulating the syntax tree
+
+Gazelle uses the [`build` package](https://pkg.go.dev/github.com/bazelbuild/buildtools/build) (from buildifier and buildozer) to parse, edit, and format `BUILD` files. This package provides a low-level interface to the syntax tree. For more convenient editing and merging, Gazelle provides its own [`merger`](https://pkg.go.dev/github.com/bazelbuild/bazel-gazelle/merger) and [`rule`](https://pkg.go.dev/github.com/bazelbuild/bazel-gazelle/rule) packages.
+
+The `rule` package lets extensions create, update, and delete "rules" (which become call expressions in the underlying syntax tree) and read or write their attributes using simple values rather than syntax tree nodes. For example, an extension can create a new rule as follows:
```

Comment:
Should we add a paragraph about how this nomenclature differs from Bazel end user documentation (but aligns with Bazel's Java code)? This would be a new target of the `go_binary` rule, not a new rule.